ui/jobs/jobs.js

```diff
@@ -2915,6 +2940,119 @@ class Bars {
     });
   }
 
+  setupDnc() {
+    const comboTimer = this.addTimerBar({
+      id: 'dnc-timers-combo',
+      fgColor: 'dnc-color-combo',
+    });
+    this.comboFuncs.push((skill) => {
+      comboTimer.duration = 0;
+      if (this.combo.isFinalSkill)
+        return;
+      if (skill)
+        comboTimer.duration = 15;
+    });
+
+    const standardStep = this.addProcBox({
+      id: 'dnc-procs-standardstep',
+      fgColor: 'dnc-color-standardstep',
+    });
+    this.abilityFuncMap[kAbility.StandardStep] = () => {
+      standardStep.duration = 0;
+      standardStep.duration = 30;
+    };
+
+    // TechnicalStep cooldown on begin dance, but effect appear when TechnicalFinish.
+    const technicalStep = this.addProcBox({
+      id: 'dnc-procs-technicalstep',
+      fgColor: 'dnc-color-technicalstep',
+    });
+    this.abilityFuncMap[kAbility.TechnicalStep] = () => {
+      technicalStep.duration = 0;
+      technicalStep.duration = 120;
+    };
+    let technicalFlag = false;
+    let elapsed = 0;
+    this.abilityFuncMap[kAbility.TechnicalFinish] = () => {
+      // Avoid mutiple call in one TechnicalFinish.
+      if (!technicalFlag) {
+        elapsed = technicalStep.elapsed;
+        technicalFlag = true;
+      }
+      technicalStep.duration = 20;
+      technicalStep.threshold = 1000;
+      technicalStep.fg = computeBackgroundColorFrom(technicalStep, 'dnc-color-technicalstep.active');
+      setTimeout(() => {
+        technicalFlag = false;
+        technicalStep.duration = 100 - elapsed;
+        technicalStep.threshold = this.gcdSkill() + 1;
+        technicalStep.fg = computeBackgroundColorFrom(technicalStep, 'dnc-color-technicalstep');
+      }, 20000);
```

Comment:
Where do these `20000` values come from? Should this be something like `technicalStep.duration * 1000`?

Reply:
It means simply 20 seconds, as the duration of these raidbuffs.
I think your suggestion is aim for easier update this block when buff time changed in later patch?
Seems great.